### ISSUE TITLE
Improved validation

### ISSUE THREAD

**Issue by AshleyEke** [auth0/terraform-provider-auth0#660]
### Checklist

- [X] I agree to the terms within the [Auth0 Code of Conduct](https://github.com/auth0/open-source-template/blob/master/CODE-OF-CONDUCT.md).

### Describe the problem you'd like to have solved

I have been working with the auth0_client resource for the first time and I'm finding a lot of issues with my configuration are only being flagged when I do a terraform apply. For basic validation rules it feels like these should be flagged when I do a terraform validate. This way it would be caught a lot earlier by my CI/CD pipeline.

### Describe the ideal solution

Basic validation rules should be flagged in the validate stage.

For example, we had an issue where the mobile configuration was not set for an app, but was still adding an empty block to the plan i.e.
```
+ mobile {
          + android {
              + app_package_name         = ""
              + sha256_cert_fingerprints = []
            }

          + ios {
              + app_bundle_identifier = ""
              + team_id               = ""
            }
        }
```

This was producing the following error at apply time:
```
Error: 400 Bad Request: Payload validation error: 'String is too short (0 chars), minimum 1' on property mobile.ios.app_bundle_identifier (Assigned by the developer to the app as its unique identifier inside the store, usually is a reverse domain plus the app name: <code>com.you.MyApp</code>). (also) Payload validation error: 'String is too short (0 chars), minimum 1' on property mobile.ios.team_id (Identifier assigned to the account that signs and upload the app to the store).
```

The error is because if you define the mobile block, the mobile properties become required. This feels like such a simple validation check that I would expect it to be flagged when running `terraform validate`

**Comment by sergiught** [comment]
Hey @AshleyEke,

Thanks for taking the time to raise this with us. I agree with your sentiment, having the validation errors raised earlier even before the API calls happen is a much better experience. We'll look into improving this for the upcoming v1 release. We'll keep the issue open and circle back once we have an update. Appreciate your patience! Please keep the feedback coming if you notice any other parts that can be improved.

**Comment by sergiught** [comment]
Hey @AshleyEke     👋🏻 

This should be fixed now as part of the [v1.0.0-beta.0](https://registry.terraform.io/providers/auth0/auth0/latest/docs) release! 🙌🏻 Please give it a try and let us know if you encounter any issues.

```
terraform {
  required_providers {
    auth0 = {
      source  = "auth0/auth0"
      version = "1.0.0-beta.0"
    }
  }
}
```

### 📚 Resources

[Quickstart](https://registry.terraform.io/providers/auth0/auth0/latest/docs/guides/quickstart) | [Documentation](https://registry.terraform.io/providers/auth0/auth0/latest/docs) | [Migration Guide](https://github.com/auth0/terraform-provider-auth0/blob/v1/MIGRATION_GUIDE.md) | [Changelog](https://github.com/auth0/terraform-provider-auth0/releases/tag/v1.0.0-beta.0)
